asign1.c

```
@@ -0,0 +1,107 @@
+
+Output Format:
+- A single integer or an error message depending on the evaluation result.
+
+Test Cases:
+1. Input: "3 + 5 * 2"
+   Output: 13
+
+2. Input: "3 + a * 2"
+   Output: Error: Invalid expression
+*/
+
+
+
+#include<stdio.h>
+#include<ctype.h>
+#include <string.h>
+int main(){
+    char str[1000];
+    printf("enter the string of expression using only +,-,*,/ operators:\n");
+    fgets(str, sizeof(str), stdin);
+    str[strcspn(str, "\n")] = 0; 
+
+    int p=strlen(str);
+    int curr=0;
+    int last = 0;
+    
+    char sign = '+';
+    int result = 0;
+    for (int i = 0; i < p;i++){ 
+        char chr = str[i];
+        if(isdigit(chr)){
+            curr = curr * 10 + chr - '0';
+        }
+        if((!isdigit(chr) && chr != ' ') || i == p- 1){
+            if(chr !='+'  && chr!='-' && chr!='*' && chr!='/'&& !isdigit(chr) && !isspace(chr)){
+                printf("Error: Invalid expression.\n");
+                return 0;
+            
+            
+                
+            }
+            if (i == p - 1 && !isdigit(chr)) {
+                printf("Error: Invalid expression.\n");
+                return 0;
+            }
+            
+            if(sign=='+'){
+                result = result + last;
+                last = curr;
+            }
+            else if(sign=='-'){
+                result = result + last;
+                last = -curr;
+            }
+            else if(sign=='*'){
+                last = last * curr;
+            }
+            else if(sign=='/'){
+                if(curr==0){
+                    printf("Error: Division by zero.\n");
+                    return 0;
+                     
+                }
+                last = last / curr;
+            }
+            
+            sign = chr;
+            curr = 0;
+                
+            
+        }
+       
+       
+    }
+    result=result+last;
+    printf("%d\n", result);
+    return 0;
+}
```

You can segregate the logic into a function and then use that function for better readability 

Additionally, when given this input - 3 ++ 2 
Code fails silently with no error message

---

segregate the logic into a function in that part i am currently working